src/System.Text.Encoding/tests/NegativeEncodingTests.cs

```diff
@@ -168,10 +181,18 @@ public static void GetString_Invalid(Encoding encoding)
             // Bytes is null
             Assert.Throws<ArgumentNullException>("bytes", () => encoding.GetString(null));
             Assert.Throws<ArgumentNullException>("bytes", () => encoding.GetString(null, 0, 0));
-            
+
             // Index or count < 0
-            Assert.Throws<ArgumentOutOfRangeException>("index", () => encoding.GetString(new byte[1], -1, 0));
-            Assert.Throws<ArgumentOutOfRangeException>("count", () => encoding.GetString(new byte[1], 0, -1));
+            if (encoding is ASCIIEncoding)
+            {
+                Assert.Throws<ArgumentOutOfRangeException>("byteIndex", () => encoding.GetString(new byte[1], -1, 0));
+                Assert.Throws<ArgumentOutOfRangeException>("byteCount", () => encoding.GetString(new byte[1], 0, -1));
+            }
+            else
+            {
+                Assert.Throws<ArgumentOutOfRangeException>("index", () => encoding.GetString(new byte[1], -1, 0));
+                Assert.Throws<ArgumentOutOfRangeException>("count", () => encoding.GetString(new byte[1], 0, -1));
+            }
```

Comment:
As noted in another comment, we shouldn't duplicate the entire statements, and should instead simply make the argument name conditional.